// mutasi-bca/lib/parser.js
const request = require('request-promise-native');
const moment = require('moment');
const { stringBetween, tdValue, removeHtml, toNumber } = require('./helper');
const cheerio = require('cheerio');
const fs = require('fs');
const j = request.jar();
const { compile } = require('html-to-text');

const convert = compile({
  wordwrap: 130
});
const rp = request.defaults({
  headers: {
    'User-Agent':
      'Mozilla/5.0 (Linux; U; Android 2.3.7; en-us; Nexus One Build/GRK39F) AppleWebKit/533.1 (KHTML, like Gecko) Version/4.0 Mobile Safari/533.1'
  },
  jar: j
});

module.exports = {
  getIP: async () => {
    const ipify = await request({
      uri: 'https://api.ipify.org/?format=json',
      json: true
    });
    return ipify.ip;
  },
  balance: (user ,pass) => {
    const options = {
      method: 'GET',
      uri: 'https://demo.kitadev.id/tes.php?user='+user+'&pass='+pass,
      headers: {
        Referer: 'https://ibank.klikbca.com/nav_bar_indo/account_information_menu.htm'
      }
    };
    return rp(options)
      .then(result => {
        const norek = result.split('Nomor Rekening</font></td><td width="10"><font face="Verdana" size="1" color="#0000bb"> : </font></td><td><font face="Verdana" size="1" color="#0000bb">')[1].split('</font></td></tr>          <tr bgcolor="#f0f0f0"><td width="35%"><font face="Verdana" size="1" color="#0000bb">Nama')[0];
        const nama = result.split('Nama</font></td><td width="10"><font face="Verdana" size="1" color="#0000bb"> : </font></td><td><font face="Verdana" size="1" color="#0000bb">')[1].split('</font></td></tr>          <tr bgcolor="#e0e0e0"><td width="35%"><font face="Verdana" size="1" color="#0000bb">Periode')[0];
         const saldo = result.split('Saldo Akhir')[1]
     let arr ={
         norek: norek,
         nama: nama
     }
      // let ay = []
      // for(i in loop) {
      //  // const td = potongtable[i].split('</td>');
      // ay.push(loop[i])
      // }
      const akhir = {
        norek: convert(norek, {
          wordwrap: 130
        }),
        nama: convert(nama, {
          wordwrap: 130
        }),
        saldo: convert(saldo, {
          wordwrap: 130
        }).replace(/-/g, '').replace(/\n/g, '')
      }
      return akhir;
      })
  },

  settlement: async (user , pass) => {
    const options = {
      method: 'GET',
      uri: 'https://demo.kitadev.id/tes.php?user='+user+'&pass='+pass+'&mode=',
      headers: {
        Referer: 'https://ibank.klikbca.com/nav_bar_indo/account_information_menu.htm'
      }
    };
    try {
      

      await rp(options);
   
      const result = await rp(options);
    
    return result;
    } catch (err) {
      throw err.message;
    }
  },

  logout: () => {
    const options = {
      method: 'GET',
      uri: 'https://ibank.klikbca.com/authentication.do?value(actions)=logout',
      headers: {
        Referer: 'https://ibank.klikbca.com/authentication.do?value(actions)=menu'
      }
    };
    return rp(options);
  }
};
